canonical/grafana-agent-operator
# Add charm tracing

## Issue
Add charm tracing to grafana-agent.

## Context
We've already updated the protocol to be ready for CMR consumption, however we're still working on ingressed communication with tempo receivers. This PR adds the groundwork needed in grafana-agent to create and send traces, however we'll still need Tempo to receive traces through ingress. This work is done in parallel.


## Testing Instructions
Pack and start grafana-agent on a machine model. In another, k8s model, start tempo-k8s and create an offer to be consumed by the machine model. See that `grafana-agent` is able to relate to that offer.
